pkg/utils/docker.go

```diff
@@ -185,6 +188,51 @@ func PullImage(image string, jsonOutput bool) {
 	}
 }
 
+// ValidateImageDigest - will ensure the image digest matches that of the one in dockerhub
+// returns imageID, docker error
+func ValidateImageDigest(image string) (string, *DockerError) {
+
+	ctx := context.Background()
+	cli, err := client.NewClientWithOpts(client.WithVersion("1.30"))
+	errors.CheckErr(err, 200, "")
+	// call docker api for image digest
+	queryDigest, err := cli.DistributionInspect(ctx, image, "")
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	// turn digest -> []byte -> string
+	digest, _ := json.Marshal(queryDigest.Descriptor.Digest)
+	fmt.Println("Query image digest is.. ", queryDigest.Descriptor.Digest)
+
+	// get local image digest
+	imageList := GetImageList()
+	imageName := strings.TrimPrefix(image, "docker.io/")
+	imageArr := []string{
+		imageName,
+	}
+
+	for _, image := range imageList {
+		imageRepo := strings.Join(image.RepoDigests, " ")
+		imageTags := strings.Join(image.RepoTags, " ")
+		for _, index := range imageArr {
+			if strings.Contains(imageTags, index) {
+				if strings.Contains(imageRepo, strings.Replace(string(digest), "\"", "", -1)) {
+					length := len(strings.Replace(string(digest), "\"", "", -1))
+					last10 := strings.Replace(string(digest), "\"", "", -1)[length-10 : length]
+					fmt.Printf("Validation for image digest ..%v succeeded\n", last10)
+					return "", nil
+				} else {
+					logr.Warnln("Local image digest did not match queried image digest from dockerhub - This could be a result of a bad download")
```

Comment:
Would this and other log output break the --json global flag that should only display JSON formatted messages ?

Reply:
changed to trace